app/Models/Transaction.php

```diff
@@ -220,6 +231,15 @@ public function vendorField(): string|null
         return $this->vendorFieldContent;
     }
 
+    public function fee(): BigNumber
+    {
+        if ($this->receipt === null) {
+            return $this->gas_price;
+        }
+
+        return $this->gas_price->multipliedBy($this->receipt->gas_used->valueOf());
```

Comment:
other issue with this is that `multipliedBy` assigns the value to the variable as well, so when it gets called multiple times in a row it ends up multiplying the result value by the `gas_used` value over and over, resulting in a larger value each time